pkg/capabilities/capabilities.go

```diff
@@ -83,12 +83,12 @@ type CapabilityRequest struct {
 }
 
 type TriggerEvent struct {
+	// The ID of the trigger capability
 	TriggerType string
-	ID          string
-	Timestamp   string
-	// Trigger-specific payload+metadata
-	Metadata values.Value
-	Payload  values.Value
+	// The ID of the trigger event
+	ID string
+	// Trigger-specific payload
+	Outputs *values.Map
```

Comment:
What's the benefit of this change? You're adding one more layer of nesting: previously it was Event->Payload and now it is Event->Outputs->Payload. In the PR description you say "This is important for compatibility with other parts of the API." - can you explain it please?

Reply:
Sure: for interpolation to work correctly we need to be passing a single values.Map around from step to step; we previously worked around this by wrapping this TriggerEvent in a map and then shipping this to the next step.

However, the TriggerType + ID should really be part of the typed API, which would have meant that we'd have to combine the Metadata and Payload before passing it to the next step (and handling cases where they aren't maps too).

I don't think we've actually changed the amount of nesting with this change:
* previously we had CapabilityResponse -> Value -> (unwrap TriggerEvent) -> (unwrap Payload) -> []FeedReport 
* now we have TriggerResponse -> Event -> Outputs -> (unwrap StreamsTriggerPayload) -> []FeedReport

This will largely not matter for capabilities -- they'll receive response.Event.Outputs directly. BTW we could reduce the nesting by one level by making the TriggerResponse:

```
type TriggerResponse struct {
    TriggerType string
     ID string
     Outputs *values.Map
     Err error
}
```

but that feels a bit less clean to me? Not opposed to it though

Reply:
Yes can do 👍 @bolekk Are you happy with that?

Reply:
sorry, what's the proposal?

Reply:
After chatting offline I'm OK with keeping stuff as-is.